test/src/e2e/multisig.test.ts

```diff
@@ -39,12 +42,39 @@ import {
     daveSecret,
     faucetAddress
 } from "../helper/constants";
+import { AssetTransaction } from "codechain-sdk/lib/core/Transaction";
 import CodeChain from "../helper/spawn";
+import { ChainRpc } from "codechain-sdk/lib/rpc/chain";
 
 const { PUSH, PUSHB, CHKMULTISIG } = Script.Opcode;
 
+// If one only sends certainly failing trasactions, the miner would not generate any block.
+// So to clearly check the result failed, insert the failing transactions inbetween succeessful ones.
+async function sendFailingTx(
+    node: CodeChain,
+    targetTx: Transaction & AssetTransaction
+): Promise<H256> {
+    await node.sdk.rpc.devel.stopSealing();
+
+    const blockNumber = await node.getBestBlockNumber();
+    const seq = await node.sdk.rpc.chain.getSeq(faucetAddress);
+    const signedDummyTx = await node.sendPayTx({ seq, quantity: 1 });
+    const targetTxHash = await node.sendAssetTransaction(targetTx, {
+        seq: seq + 1
+    });
+
+    await node.sdk.rpc.devel.startSealing();
+    await node.waitBlockNumber(blockNumber + 1);
+
+    expect(await node.sdk.rpc.chain.containsTransaction(signedDummyTx.hash()))
+        .be.true;
+
+    return targetTxHash;
```

Comment:
Is the caller should know the transaction hash?
If all callers use the hash only to check whether the transaction failed, I think it can be included in this function.

Reply:
At first I was considering including the check routine for the transaction failure, but soon after I thought it is natural for all test functions included in "it" marked test cases to have explicit "expect" in their bodies.
I thought it looks consistent that all test cases marked with "it" end with the "expect" clauses.
Does the become better after including the check routines?

Reply:
I think it's better to rename this function to `expectTransactionFail` and to check the failure internally.

Reply:
I renamed and checked the failure inside the function.